aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -235,6 +235,28 @@ Tensor& mvlgamma_(Tensor& self, int64_t p) {
   return self.copy_(args.lgamma_().sum(-1).add_(p * (p - 1) * std::log(M_PI) / 4.));
 }
 
+Tensor& cumsum_out_cpu(Tensor& result, const Tensor& self, int64_t dimension) {
+  int64_t dim = maybe_wrap_dim(dimension, self.dim());
+  auto device1 = self.type().device_type();
+  TORCH_CHECK(device1 == kCPU || device1 == kCUDA, "cumsum only supports CPU and CUDA devices, self got: ", device1);
+  auto device2 = result.type().device_type();
+  TORCH_CHECK(device2 == kCPU || device2 == kCUDA, "cumsum only supports CPU and CUDA devices, result got: ", device2);
+  TORCH_CHECK(device1 == device2, "self and result must have the same device type. self: ", device1, " result: ", device2);
+  TORCH_CHECK(!self.is_cuda() || self.get_device() == result.get_device(), "device of self (", self.get_device(), ") must match device of result (", result.get_device(), ")");
+  if (self.dim() < 1) {
+    result.copy_(self);
+  } else {
+    cumsum_stub(device1, result, self, dim);
+  }
+  return result;
+}
+
+Tensor cumsum_cpu(const Tensor& self, int64_t dimension) {
+  Tensor result = at::empty_like(self);
```

Comment:
Should we use the input dtype or better go for an accumulator-type?
e.g. the old behavior was
```
>>> x = torch.randint(0, 255, (100,), dtype=torch.int8)
>>> x.type()
'torch.CharTensor'
>>> x.cumsum(0).type()
'torch.LongTensor'
```

Reply:
@andreaskoepf thanks you for you comments, this PR still in progress. It's draft and has [WIP] (work in progress) tag.
I think you are right, accscalar_t is a good idea.